packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -54,6 +55,16 @@ class ModalBarrier extends StatelessWidget {
   ///    [ModalBarrier] built by [ModalRoute] pages.
   final bool dismissible;
 
+  /// Called when the barrier is being dismissed.
+  ///
+  /// If non-null [onDismiss] will be called in place of popping the current
+  /// route. It is up to the callback to handle dismissing the barrier.
+  ///
+  /// If null, the ambient [Navigator]'s current route will be popped.
+  ///
+  /// This field is ignored if [dismissible] is false.
+  final VoidCallback? onDismiss;
```

Comment:
I am not entirely sure how this relates to the AutoComplete widget, I assume there is another pr to hook this up to the autocomplete?

Reply:
I wonder if we can use addLocalHistoryEntry to handle this? for example, the autocomplete will add a local history entry to the modal route, and the pop will pop the local history entry before popping the route

Reply:
Oh, that's actually a good point. What does the integration in the AutoComplete widget look like? Maybe it should be a local route that's getting dismissed with the modal barrier!

Reply:
Yeah, this was a standalone PR I pulled out of some Autocomplete work as it wasn't specific to the Autocomplete widget.

Currently the Autocomplete widget just manages an overlay entry to display the options and doesn't involve routes at all. The reason I was looking at reusing the ModalBarrier is that it already had support for the proper touch handling and semantics, which I didn't want to reimplement just for Autocomplete.

I wasn't aware of the local history mechanism. I will look into that, and see if it makes more sense. Thanks for the pointer.